packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/Serialization/TypeSpecInputModelTypeConverter.cs

```diff
@@ -33,7 +33,21 @@ public static InputModelType CreateModelType(ref Utf8JsonReader reader, string?
             id = id ?? throw new JsonException();
 
             // create an empty model to resolve circular references
-            var model = new InputModelType(name!, null!, null, null, null, InputModelTypeUsage.None, null!, null, null!, null, null, null!, null, false);
+            var model = new InputModelType(
+                name: name!,
+                crossLanguageDefinitionId: null!,
+                access: null,
+                deprecation: null,
+                description: null,
+                usage: InputModelTypeUsage.None,
+                properties: null!,
+                baseModel: null,
+                derivedModels: new List<InputModelType>(),
```

Comment:
Instead of the cast we can have an internal property which exposes List

![image](https://github.com/user-attachments/assets/56fdbb6c-01cb-4594-8051-3c7af6b56a92)